src/app/console.py

```diff
@@ -1,58 +1,188 @@
+        else:
+            self.zemlja.hero.premikanje(dx=dx, dy=dy)
+
+    def premakni_barabo(self):
+
+        trenutnePozicijeBarab = []
 
         for b in self.zemlja.barabe:
+            trenutnePozicijeBarab.append((b.x, b.y))
+
+        for b in self.zemlja.barabe:
+            trenutnePozicijeBarab.remove((b.x, b.y))
             b.nakljucno_gibanje()
-#  implement end game detection logic
\ No newline at end of file
+            x, y = b.x, b.y
+            # če naslednja naključna pozicija barabe že obstaja, ponovno izneri pozicijo
+            while (x, y) in trenutnePozicijeBarab:
+                b.nakljucno_gibanje()
+                x, y = b.x, b.y
+            trenutnePozicijeBarab.append((x, y))
+
+    def end_game_loose(self):
+        xp, yp = self.zemlja.princeska.x, self.zemlja.princeska.y
+        xh, yh = self.zemlja.hero.x, self.zemlja.hero.y
+
+
+        for b in self.zemlja.barabe:
+            xb, yb = b.x, b.y
+            # pogoji za konec igre
+            if yh == yb and xb in range(xh - 1, xh + 2):
+                print('\033[31mHeroj ujet!Konec igre...\033[0m')
+                print(f'\033[92mVaš končni SCORE: {self.rezultat} in prišli ste do LEVEL: {self.level}\033[0m')
+                return True
+            elif xh == xb and yb in range(yh - 1, yh + 2):
+                print('\033[31mHeroj ujet!Konec igre...\033[0m')
+                print(f'\033[92mVaš končni SCORE: {self.rezultat} in prišli ste do LEVEL: {self.level}\033[0m')
+                return True
+            elif xb == xp and yb == yp:
+                print('\033[31mBaraba ujela princesko!Konec igre...\033[0m')
+                print(f'\033[92mVaš končni SCORE: {self.rezultat} in prišli ste do LEVEL: {self.level}\033[0m')
+                return True
+            elif self.maxKoraki == 0:
+                print('\033[31mPresegli ste določeno število potez, igre konec!\033[0m')
+                print(f'\033[92mVaš končni SCORE: {self.rezultat} in prišli ste do LEVEL: {self.level}\033[0m')
+                return True
+
+    def end_game_win(self):
+        xh, yh = self.zemlja.hero.x, self.zemlja.hero.y
+        xp, yp = self.zemlja.princeska.x, self.zemlja.princeska.y
+
+        if xh == xp and yh == yp:
+            self.level += 1
+            self.rezultat += self.maxKoraki * 2
+            if self.level > 7:
+                print(
+                    f'\033[33mČestitke, končali ste igro, princesa in heroj živita srečno skupaj vse do konca svojih dni!\033[0m')
+                print(f'\033[92mVaš končni SCORE: {self.rezultat} in prišli ste do LEVEL: {self.level}\033[0m')
+                return True
+            else:
+                # bug : če pride do princeske ne premikaj več barab, oziroma draw new game()
+                print('\033[32mPrinceska rešena...greš v naslednji nivo\033[0m')
+                return False
+
+
+# prestavil importe po izvajanju kode zaradi errorja 'circular import', kjer se moduli kličejo v loopu...
```

Comment:
Tipična napaka začetnikov...
* Domenski prostor ne sme vedeti v kateri aplikaciji se domena izvaja!!!
* Circular dependency se ustvari ker importaš v domeni consolo kar ni prav, saj tako domena dobi dependency na consolo.

Če pravilno ustvariš arhitekturo potem ni šans da dobiš circular dependenci in ne ustvarjat workaround-e ki rešijo circular dependenci, popraviti moraš arhitekturo pa bo sama od sebe izginila...